api/include/opentelemetry/context/runtime_context.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include "opentelemetry/context/context.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+// Provides a wrapper for propagating the context object globally. In order
+// to use either the threadlocal_context.h file must be included or another
+// implementation which must be derived from the RuntimeContext can be
+// provided.
+class RuntimeContext
+{
+public:
+  class Token
```

Comment:
Would it make sense to have a dtor for Token which calls Detach?

Reply:
That make sense, seems like it will add an element of safety so the context can't get stuck after a token has gone out of scope. I will add it.

Reply:
Added.